app/src/ui/lib/avatar.tsx

```diff
@@ -39,6 +39,81 @@ const avatarTokenCache = new ExpiringOperationCache<
   () => offsetFrom(0, 50, 'minutes')
 )
 
+const getBotLogin = (user: IAvatarUser) => {
+  const { endpoint } = user
+  if (user.avatarURL !== undefined || endpoint === null) {
+    return undefined
+  }
+
+  const match = parseStealthEmail(user.email, endpoint)
+
+  return match?.login?.endsWith('[bot]') ? match.login : undefined
+}
+
+const botAvatarCache = new ExpiringOperationCache<
+  { user: IAvatarUser; accounts: ReadonlyArray<Account> },
+  IAvatarUser
+>(
+  ({ user }) => `${user.endpoint}:${user.email}`,
+  async ({ user, accounts }) => {
+    const { endpoint } = user
+    if (user.avatarURL !== undefined || endpoint === null) {
+      throw new Error('Avatar URL already resolved or endpoint is missing')
+    }
+
+    const account = accounts.find(a => a.endpoint === user.endpoint)
+
+    if (!account) {
+      throw new Error('No account found for endpoint')
+    }
+
+    const login = getBotLogin(user)
+
+    if (!login) {
+      throw new Error('Email does not appear to be a bot email')
+    }
+
+    const api = new API(endpoint, account.token)
+    const apiUser = await api.fetchUser(login)
+
+    if (!apiUser?.avatar_url) {
+      throw new Error('No avatar url returned from API')
+    }
+
+    return { ...user, avatarURL: apiUser.avatar_url }
+  },
+  ({ user }) =>
+    user.endpoint && isGHE(user.endpoint)
+      ? offsetFrom(0, 50, 'minutes')
+      : Infinity
+)
+
+const dotComBot = (login: string, id: number, integrationId: number) => {
+  const avatarURL = `https://avatars.githubusercontent.com/in/${integrationId}?v=4`
+  const endpoint = getDotComAPIEndpoint()
+  const stealthHost = 'users.noreply.github.com'
+  return [
+    {
+      email: `${id}+${login}@${stealthHost}`,
+      name: login,
+      avatarURL,
+      endpoint,
+    },
+    { email: `${login}@${stealthHost}`, name: login, avatarURL, endpoint },
+  ]
+}
+
+const knownAvatars: ReadonlyArray<IAvatarUser> = [
+  ...dotComBot('dependabot[bot]', 49699333, 29110),
+  ...dotComBot('github-actions[bot]', 41898282, 15368),
+  ...dotComBot('github-pages[bot]	', 52472962, 34598),
```

Comment:
Curious why the `github-pages[bot]` has space after it's login?

Reply:
🤯 and it's a tab, not just a bunch of spaces

I didn't notice it, nice catch! It might have been an issue interacting with the IDE for autocompletion / Copilot suggestions. I'll remove it. Thanks 🙏